pyannote/audio/applications/change_detection.py

```diff
@@ -0,0 +1,420 @@
+    mkdir_p(store_dir)
+
+    # -- LOAD MODEL --
+    architecture_yml = train_dir + '/architecture.yml'
+    WEIGHTS_H5 = train_dir + '/weights/{epoch:04d}.h5'
+    nb_epoch = 0
+    while True:
+        weights_h5 = WEIGHTS_H5.format(epoch=nb_epoch)
+        if not os.path.isfile(weights_h5):
+            break
+        nb_epoch += 1
+    config_dir = os.path.dirname(os.path.dirname(train_dir))
+    config_yml = config_dir + '/config.yml'
+    with open(config_yml, 'r') as fp:
+        config = yaml.load(fp)
+
+    # -- FEATURE EXTRACTION --
+    feature_extraction_name = config['feature_extraction']['name']
+    features = __import__('pyannote.audio.features',
+        fromlist=[feature_extraction_name])
+    FeatureExtraction = getattr(features, feature_extraction_name)
+    feature_extraction = FeatureExtraction(
+            **config['feature_extraction'].get('params', {}))
+
+    # -- SEQUENCE GENERATOR --
+    duration = config['sequences']['duration']
+    step = config['sequences']['step']
+
+    groundtruth = {}
+    for dev_file in getattr(protocol, subset)():
+        uri = dev_file['uri']
+        groundtruth[uri] = dev_file['annotation']
+
+    def objective_function(epoch):
+        weights_h5 = WEIGHTS_H5.format(epoch=epoch)
+        sequence_labeling = SequenceLabeling.from_disk(
+                architecture_yml, weights_h5)
+
+        aggregation = SequenceLabelingAggregation(
+                sequence_labeling, feature_extraction,
+                duration=duration, step=step)
+
+
+        predictions = {}
+
+        for dev_file in getattr(protocol, subset)():
+            uri = dev_file['uri']
+            predictions[uri] = aggregation.apply(dev_file)
+
+        alphas = np.linspace(0, 1, 20)
+
+        purity = [SegmentationPurity(parallel=False) for alpha in alphas]
+        coverage = [SegmentationCoverage(parallel=False) for alpha in alphas]
+
+        for i, alpha in enumerate(alphas):
+            # initialize peak detection algorithm
+            peak = Peak(alpha=alpha, min_duration=min_duration)
+            for uri, reference in groundtruth.items():
+                # apply peak detection
+                hypothesis = peak.apply(predictions[uri])
+                # compute purity and coverage
+                purity[i](reference, hypothesis)
+                coverage[i](reference, hypothesis)
+
+        TEMPLATE = '{alpha:g} {purity:.3f}% {coverage:.3f}%'
+        res = []
+        for i, a in enumerate(alphas):
+            p = 100 * abs(purity[i])
+            c = 100 * abs(coverage[i])
+            print(TEMPLATE.format(alpha=a, purity=p, coverage=c))
+            res.append((a,p,c))
+        return res
+
+    if epoch is None or epoch > nb_epoch:
+        res = objective_function(nb_epoch-1)
+    else:
+        res = objective_function(epoch)
+
+    with open(store_dir + '/res.yml', 'w') as fp:
+        yaml.dump(res, fp, default_flow_style=False)
```

Comment:
Can you please tell me what this file looks like? 
Why use yaml format and not simple text file?